lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -351,4 +352,78 @@ public Classification getClassificationByTask(Task task) throws ClassificationNo
         }
     }
 
+    @Override
+    public void deleteClassification(String classificationKey, String... domains) throws ClassificationInUseException, NotAuthorizedException, ClassificationNotFoundException {
+        if (domains.length == 0) {
+            throw new IllegalArgumentException("You have to specify a domain");
+        }
+        for (String domain : domains) {
+            if (domain.equals("")) {
+                // Alternative Solution: open "deleteRootClassification".
+                // But in my opinion there is a big difference between deleting a specification of a Classification and deleting the whole classification.
+                throw new IllegalArgumentException("You cannot delete the Root Classification this way. Please use 'deleteRootClassification' instead");
+            }
+        }
+
+        TaskServiceImpl taskService = (TaskServiceImpl) taskanaEngineImpl.getTaskService();
+        boolean existingTasksForClassificationInDomains = ((TaskServiceImpl) taskService).isThereATaskForThisClassificationKeyInTheseDomains(classificationKey, domains);
```

Comment:
Other Entities/Models/... got Classifications, too. Sry, but you need to change the name of the variable and also the query to test it. 
Just COUNT references and give back a number >0 = there are references and you can´t delete. This would be readable.

Reply:
Please move this to a method in ClassificationService and just use a TaskQuery. This method could throw the exception right away...